custom/icds/location_rationalization/parser.py

```diff
@@ -95,5 +98,21 @@ def _note_transition(self, operation, location_type, new_site_code, old_site_cod
             self.transitions[location_type][operation][new_site_code] = old_site_code
         elif operation == EXTRACT_OPERATION:
             self.transitions[location_type][operation][new_site_code] = old_site_code
+        self.site_codes_to_be_archived.append(old_site_code)
         self.requested_transitions[old_site_code] = operation
         self.requested_transitions[new_site_code] = operation
+
+    def validate(self):
+        """
+        ensure all locations getting archived, also have their descendants getting archived
```

Comment:
What happens if someone uploads a file that doesn't go down to the lowest level? It looks like if someone uploads a file that contains just state + district + block, for every block that gets archived, this code will add an error for every descendant supervisor and awc, since none of those will be in `site_codes_to_be_archived`. Am I missing something?

Reply:
That wont happen. The upload form validates that all columns are present for all location types.